File: apps/frilvault-cli/src/command/stats.rs
use anyhow::Result;
use frilvault_core::FrilVault;

use crate::{
    cli::stats::StatsCommand,
    output::{OutputFormat, print_json, resolve_format},
};

pub fn execute(command: StatsCommand) -> Result<()> {
    let vault = FrilVault::open(std::env::current_dir()?)?;
    let mut service = vault.workspace()?;

    let stats = service.stats()?;

    if matches!(resolve_format(command.format), OutputFormat::Json) {
        print_json(&stats)?;
        return Ok(());
    }

    println!("Workspace Statistics\n");

    println!("Files: {}", stats.file_count,);

    println!("Total Notes: {}", stats.total_notes,);

    println!("Existing Files: {}", stats.existing_files,);

    println!("Missing Files: {}", stats.missing_files,);

    println!();

    println!("Line Notes: {}", stats.line_notes,);

    println!("Symbol Notes: {}", stats.symbol_notes,);

    Ok(())
}

File: apps/vscode-extension/src/features/workspace/stats.ts
import * as vscode from 'vscode';

import type { CliClient } from '../../core/cliClient';

export function createShowStatsCommand(
  cliClient: CliClient,
  getWorkspaceRoot: () => string,
): () => Promise<void> {
  return async () => {
    const stats = await cliClient.workspaceStats(getWorkspaceRoot());
    const channel = vscode.window.createOutputChannel('FrilVault Stats');
    channel.clear();
    channel.appendLine(`files: ${stats.file_count}`);
    channel.appendLine(`notes: ${stats.total_notes}`);
    channel.appendLine(`existing files: ${stats.existing_files}`);
    channel.appendLine(`missing files: ${stats.missing_files}`);
    channel.appendLine(`line notes: ${stats.line_notes}`);
    channel.appendLine(`symbol notes: ${stats.symbol_notes}`);
    channel.show(true);
  };
}
